### PR TITLE
docs(general): update documentation to add info about v3

### LINKED ISSUE #4198: Docs: Update Powertools documentation to add a banner about v3
### What were you searching in the docs?

Update Powertools documentation to add a banner about v3	

### Is this related to an existing documentation section?

https://docs.powertools.aws.dev/lambda/python/latest/

### How can we improve?

Update Powertools documentation to add a banner about v3	

### Got a suggestion in mind?

_No response_

### Acknowledgment

- [X] I understand the final update might be different from my proposed suggestion, or refused.

### PR DESCRIPTION
**Pull request by leandrodamascena**
**Issue number:** #4198 

### Changes

Update Powertools documentation to update versioning policy and add a banner about v3

### User experience

> Please share what the user experience looks like before and after this change

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda/python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [ ] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: docs/overrides/main.html [hunk 1]
```diff
@@ -1,5 +1,9 @@
 {% extends "base.html" %}
 
+{% block announce %}
+🚨 RFC for <a href="https://github.com/aws-powertools/powertools-lambda-python/issues/4189">Powertools for AWS Lambda (Python) v3</a> is out!
```
Comment (heitorlessa) [inline docs/overrides/main.html]:
```suggestion
🚨 The next major version (v3) is coming - <a href="https://github.com/aws-powertools/powertools-lambda-python/issues/4189">come learn and discuss upcoming changes</a> is out!
```

reason: bring key info upfront, followed by call to action we want customers/community to do.

File: docs/versioning.md [hunk 2]
```diff
@@ -59,5 +59,6 @@ To see the list of available major versions of Powertools for AWS Lambda and whe
...
 | -------------------------------- | ------------- | -------------------- | ------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| Powertools for AWS Lambda Python | 2.x           | General Availability | 10/24/2022                | See [Release Notes](https://github.com/aws-powertools/powertools-lambda-python/releases/tag/v2.0.0)                                                                  |
-| Powertools for AWS Lambda Python | 1.x           | End of Support       | 06/18/2020                | See [RFC](https://github.com/aws-powertools/powertools-lambda-python/issues/1459) and [upgrade guide](https://docs.powertools.aws.dev/lambda/python/latest/upgrade/) |
+| Powertools for AWS Lambda (Python) | 3.x           | Developer Preview    |                           | See [RFC](https://github.com/aws-powertools/powertools-lambda-python/issues/4189)                                                                  |
```
Comment (heitorlessa) [inline docs/versioning.md:62]: we need to let customers know how to build/test this developer preview in the RFC later, plz.

**Review by heitorlessa — CHANGES_REQUESTED** [review]
minor change to improve banner call out; all good otherrwise!

File: docs/overrides/main.html (hunk 1 above)
Comment (leandrodamascena) [reply docs/overrides/main.html]: Make much more sense! Changed!

File: docs/versioning.md (hunk 2 above)
Comment (leandrodamascena) [reply docs/versioning.md:62]: Ok ok, noted!

**Comment by leandrodamascena** [comment]
Thanks for the review @heitorlessa! It's ready to another review.

**Review by heitorlessa — APPROVED** [review]

**Pull request merged**